docs/modules/ROOT/pages/servlet/authentication/session-management.adoc

```diff
@@ -267,6 +268,43 @@ class LoginRequest {
 And that's it.
 If you are not sure what `securityContextHolderStrategy` is in the above example, you can read more about it in the <<use-securitycontextholderstrategy, Using `SecurityContextStrategy` section>>.
 
+[[properly-clearing-authentication]]
+=== Properly Clearing an Authentication
+
+If you are using Spring Security's xref:servlet/authentication/logout.adoc[Logout Support] then it takes care of clearing the authentication for you.
+But if that is not the case, and you are manually logging users out of your application for example, you have to also be aware of how to properly clear it from the `SecurityContextRepository`.
+
+You might already be clearing the `SecurityContextHolder` by doing `SecurityContextHolderStrategy#clearContext()`, however, if your application xref:migration/servlet/session-management.adoc#_require_explicit_saving_of_securitycontextrepository[requires explicit save] of the context this is not enough because it does not remove it from the `SecurityContextRepository`, therefore it is extremely likely that the `Authentication` will still be available for the next requests.
+
+To make sure the authentication is properly cleared you have to save an empty `SecurityContext` into the `SecurityContextRepository`.
+Most implementations of `SecurityContextRepository`, like `HttpSessionSecurityContextRepository` will clear the authentication if you save an empty `SecurityContext`.
+To achieve that, you can do like so:
+
+====
+.Java
+[source,java,role="primary"]
+----
+private SecurityContextRepository securityContextRepository =
+        new HttpSessionSecurityContextRepository();
+
+private SecurityContextHolderStrategy securityContextHolderStrategy =
+        SecurityContextHolder.getContextHolderStrategy();
+
+@PostMapping("/my-logout")
+public void manualLogout(HttpServletRequest request, HttpServletResponse response) {
+    this.securityContextHolderStrategy.clearContext(); <1>
+    SecurityContext emptyContext = this.securityContextHolderStrategy.createEmptyContext(); <2>
```

Comment:
The other concern here is that this logout isn't robust since it doesn't do the other logout actions like clearing the `CsrfTokenRepository`. If we are going to show a manual logout sample, it needs to show all the things that happen by default in Spring Security so they get equivalent behavior.

Otherwise, it may be that this sample needs to become a snippet that only demonstrates clearing both the context and the repository.